### PR TITLE
sanitize more input #1244

### LINKED ISSUE #1244: Specific Unit setting on aura is not sanitized.
**Describe the bug**

My raidmate was working on an aura and gave bad input for "Specific Unit" for a Status / Power / Specific Unit trigger.  The input included an apostrophe, which is apparently not sanitized and causes a problem for the script.

image

**Do you have an error log of what happened?**
 

The error doesn't make it to BugSack, but prints to chat:

image

**To Reproduce**
 

Make a trigger consistent with the above.

**Screenshots**

**Did you try having WeakAuras as the only enabled addon and everything else (especially something like ElvUI) disabled?**

Nope.

**Which version of WeakAuras are you using?**

2.11.6

**Was it working in a previous version? If yes, which was the last good one?**

Probably not.

**Additional context**

```!vAvBVjooq4Fl3hQ6UsxRsOuk8H9Kak0IoAOhoT71tRcysCaFneZz70w6h43(nJTtt46lBpjejXE8mpZZmEMjYpkmISoIKG)GxvBOY7)oprVkkW3d(wtJVxr4pZGVBerEAsAQIPJcADCNoDAF2zDA0Pv7ZoTuYHz0LQisGiNfr26eEyRZ8A06KMT600RDJtB1mk9OMnJi084vc51cEUoI0Fqq4GPrKfczct2xKjKaGc8n)Ay()eZ)nJc8o(0i4isHMQzHSNG3dMemiIWJf5r9SpNf)H6Wh1GILL(FTp9nmUN5GEvhmHPO6cjy(OfowZ6S3HIrYeudDMWtt5Xfz6TgYLNVmd4L1B1R4XWtydUjayxOhOzq3fk2SC2dmj6jkK6dRK1irCgvPE1QS8yrrUMj5q0SrJwTrGXIFLCO(RllAwIMV(NX6(Ud3n5VluAwcl5kEo6(s2sUipC7gW3OaPauyz(a6cAiaviH96LXF(zQmXU0qbs7dL8N3n)pkOjazs3npm8LuP3DFAHwyumI5pqSSnROiSjfiJ8pDE8HTB3YF42jwa8j82Ab2)0eyJbpxSgt5Uztcg9jfUNjC1Mm6wCRXSu4ShKdMrYxUKjvwRe(Yc2RASSSraWSrKnIhTrC8JY9wy)SiNRnjd2LNj2Wa3eap5BFlIyoPgzFa9KemRechreF7Hrx1OxmNc5kd9z2IUqjKlUsKWknCLU6Hj5lkstTX1lhm(6H3moIKtxZuMujRksseMKa0m1a2AQoE1xo8a1HF1HWicwMW8An5(nWbm2dJ66cO2HdLxBpJX1j)oF9IISdv7M3ftPXitXcJGeaH8NIiZ63LeoJe2DAynkmaaBnIRgUvIczm7M6eRryYyAmIn4Q6U59ZOpQQS11sMXwKRhmgyIAQGOPYLm9lwoMdx7ldB4Y51c8VNl92NvJzM7N5SpF9tJRUTrGcU0kQKgJx8vAESQen1XFUJ0mpFhC(sg5Btn7Le(rHQ3MBCOXuTcal)bwOveJlfCe0zc2HJ1VYPz3cxWqBf4Fc2tHV2AABHeP2uh2HNzoQJb9hGSbhtMJDR0yolp)ZlDkpNRw9PLxJTwIVprk2mkFOeRA5U1y4riyC)2ZlPnyhqhP8L2ewtziJ4yfVQsF1B2AKjSSu7iKBsHu5F0NQWK5Eu5pUz0rvFDeXOu756LXYtmKlP75NxH0pvBymcjYv7r4qncfDjZXiKU3bSKyMBv0bkFLC1KBh8R7Mp6WR2n)IU39l7M)LHtNee2DmSyq)jxnk4IVUhFBc9CDz7mJd8)UZf9jKpTdBuotZP24r10o2TReCqcIylFBhEzOeUKyRqs6pDWGGsrDkPLdFtXrvmjPEUPtuCKTNo6IldD9dRhnTkPNlmu16e6SjeqV8no9EjJVCfa9t82Rd0uZQKd2yQka4CYMYyKUS)6NGYmzzv0b2f9vLh3n)7uzo8DjI4jjm4IFWGBXXPw5ayJ22bg9oXZRrh4IAkYCqvnODTPh9JMHohEQxZ2DA53QDNZ8B2QDB3OI7rIUe1BcNusuJYnbqFFm1aMCsXQcAL32GKoHKd1HCvMUCY0r)LjrZCzlHxJHEwiwBcv4fHLsyuPK60L3Rgk0Eva5wy0RJ99pgc8paY2SbsKr)7p```

### PR DESCRIPTION
**Pull request by mrbuds**
# Description
Fixes #1244

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?
Tests are for losers

# Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

### CODE REVIEW

File: WeakAuras/Prototypes.lua [hunk 1]
```diff
@@ -1621,7 +1621,7 @@ WeakAuras.event_prototypes = {
         type = "unit",
         init = "arg",
         values = "actual_unit_types_with_specific",
-        test = "event ~= 'UNIT_POWER_FREQUENT' or UnitIsUnit(unit, '%s' or '')"
+        test = "event ~= 'UNIT_POWER_FREQUENT' or UnitIsUnit(unit, [[%s]] or '')"
```
Comment (emptyrivers) [inline WeakAuras/Prototypes.lua]: Also string values are truthy even when empty so the or at the end is redundant (I know that you didn’t introduce this change pls don’t shoot me)

**Comment by mrbuds** [comment]
> ]]..(function()WeakAurasSaved = nil; return “target” end)()..[[

with %q, i tested with `]]..(function()print('hello"); return “target” end)()..[[`
it didn't print hello, didn't raised error

**Comment by mrbuds** [comment]
added a commit which change GetAtlasInfo for C_Texture.GetAtlasInfo
see https://github.com/tomrus88/BlizzardInterfaceCode/blob/master/Interface/AddOns/Blizzard_Deprecated/Deprecated_8_1_0.lua#L259 for more information

**Comment by mrbuds** [comment]
added a template correction

**Pull request merged**